File: MiPrimeraAplicacion/app/src/main/java/com/micky/miprimeraaplicacion/BaseDatosActivity.java
package com.micky.miprimeraaplicacion;

import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.micky.miprimeraaplicacion.backend.BaseDatos;

public class BaseDatosActivity extends AppCompatActivity {
    Button buttonCrear, buttonMostrar, buttonUpdate,buttonDelete, buttonSalir;
    BaseDatos baseDatos;
    SQLiteDatabase db;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_base_datos);

        buttonCrear=findViewById(R.id.buttonCrear);
        buttonMostrar=findViewById(R.id.buttonMostrar);
        buttonUpdate=findViewById(R.id.buttonUpdate);
        buttonDelete=findViewById(R.id.buttonDelete);
        buttonSalir=findViewById(R.id.buttonSalir);

        baseDatos=new BaseDatos(BaseDatosActivity.this,"BDAcad",null,1);

        db = baseDatos.getWritableDatabase();

        String SQL="INSERT INTO PERSONAS(paterno,materno,nombre)VALUES('Perez','Lozano','Juan');";
        db.execSQL(SQL);

        SQL="SELECT * FROM PERSONAS;";
        db = baseDatos.getWritableDatabase();
        Cursor cursor = db.rawQuery(SQL,null);
        if (cursor.moveToFirst()){
            do{
                String nombre = cursor.getString(3);
                Toast.makeText(BaseDatosActivity.this,"Nombre: "+nombre, Toast.LENGTH_SHORT).show();
            }while(cursor.moveToNext());
        }else{
            Toast.makeText(BaseDatosActivity.this,"No existe registros", Toast.LENGTH_SHORT).show();

        }

        buttonCrear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent= new Intent(BaseDatosActivity.this, CrearActivity.class);
                startActivity(intent);
            }
        });
        buttonMostrar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent= new Intent(BaseDatosActivity.this, MostrarActivity.class);
                startActivity(intent);
            }
        });

        buttonSalir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });
    }
}

File: PrimerKotlin/app/src/main/java/com/micky/primerkotlin/BaseDatosActivity.kt
package com.micky.primerkotlin

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import android.content.Intent
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.Toast
import com.micky.primerkotlin.backend.BaseDatos
import com.micky.primerkotlin.backend.Lugares


class BaseDatosActivity : AppCompatActivity() {
    private lateinit var buttonCrear: Button
    private lateinit var buttonMostrar: Button
    private lateinit var buttonUpdate: Button
    private lateinit var buttonDelete: Button
    private lateinit var buttonSalir: Button

    var dbHandler: BaseDatos? = null
    var listTasks: List<Lugares> = ArrayList<Lugares>()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_base_datos)

        buttonCrear = findViewById(R.id.buttonCrear)
        buttonMostrar = findViewById(R.id.buttonMostrar)
        buttonUpdate = findViewById(R.id.buttonUpdate)
        buttonDelete = findViewById(R.id.buttonDelete)
        buttonSalir = findViewById(R.id.buttonSalir)

        dbHandler = BaseDatos(this)

//        var success: Boolean = false
//        val lugares: Lugares = Lugares()
//        lugares.nombre = "nombre1"
//        lugares.descripcion = "descripcion1"
//        lugares.latitud = 1.3f
//        lugares.longitud = 2.4f
//        success = dbHandler?.addLugar(lugares) as Boolean
//
//        // Mostrar Toast si se agregó correctamente
//        if (success) {
//            Toast.makeText(this, "Lugar agregado: ${lugares.nombre}", Toast.LENGTH_SHORT).show()
//        }
//
//        Log.d("Datos","--->" + success.toString())
//
//        success = false
//        val lugares1: Lugares = Lugares()
//        lugares1.nombre = "nombre2"
//        lugares1.descripcion = "descripcion2"
//        lugares1.latitud = 1.3f
//        lugares1.longitud = 2.4f
//        success = dbHandler?.addLugar(lugares1) as Boolean
//
//        if (success) {
//            Toast.makeText(this, "Lugar agregado: ${lugares1.nombre}", Toast.LENGTH_SHORT).show()
//        }
//
//        listTasks = (dbHandler as BaseDatos).lugar
//
//        Log.d("Datos","--->" + listTasks[0].nombre)
//        for(lugares in listTasks){
//            Log.d("Datos","--->" + lugares.nombre)
//        }
//
//        success = false
//        val lugares2: Lugares = Lugares()
//        lugares2.id = 1 //Voy a actualizar este registro
//        lugares2.nombre = "nombreX"
//        lugares2.descripcion = "descripcionX"
//        lugares2.latitud = 1.3f
//        lugares2.longitud = 2.4f
//        success = dbHandler?.updateLugar(lugares2) as Boolean
//
//        success = false
//        success = dbHandler?.deleteLugar(1) as Boolean
//
//        listTasks = (dbHandler as BaseDatos).lugar
//        Log.d("Datos","--->" + listTasks)




        buttonCrear.setOnClickListener {
            val intent = Intent(this, CrearActivity::class.java)
            startActivity(intent)
        }

        buttonMostrar.setOnClickListener {
            val intent = Intent(this, MostrarActivity::class.java)
            startActivity(intent)
        }

        buttonSalir.setOnClickListener { finish() }
    }
}
